### PR TITLE
HPCC-16196 Publish correct metadata when multi-node Roxie writes files

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
Compiled

### CODE REVIEW

**Comment by richardkchapman** [comment]
@jakesmith Please review

File: roxie/ccd/ccdfile.cpp [hunk 1]
```diff
@@ -2790,14 +2790,27 @@ class CRoxieWriteHandler : implements IRoxieWriteHandler, public CInterface
 
     void copyPhysical() const
     {
+        RemoteFilename rfn, rdn;
+        dFile->getPartFilename(rfn, 0, 0);
+        StringBuffer physicalName, physicalDir, physicalBase;
+        rfn.getLocalPath(physicalName);
+        splitFilename(physicalName, &physicalDir, &physicalDir, &physicalBase, &physicalBase);
+        rdn.setLocalPath(physicalDir.str());
+        if (localCluster && getNumNodes() > 1)
```
[the remark this answers is not in the document]
Comment (richardkchapman) [reply roxie/ccd/ccdfile.cpp]: This code is copying to node n+1, which is appropriate even if on first node.

File: roxie/ccd/ccdfile.cpp [hunk 2]
```diff
@@ -2882,7 +2895,18 @@ class CRoxieWriteHandler : implements IRoxieWriteHandler, public CInterface
...
+            if (getNumNodes() > 1)
+            {
+                unsigned buddy = myNodeIndex+1;
+                if (buddy >= getNumNodes())
+                    buddy = 0;
+                SocketEndpoint buddyNode(0, getNodeAddress(buddy));
+                eps.append(buddyNode);
+            }
+            localCluster.setown(createIGroup(eps));
             localClusterName.set(cluster);
```
Comment (jakesmith) [inline roxie/ccd/ccdfile.cpp]: I think this needs to be a valid match/lookup to the ip's in the group.
So could use : eps.getText(<StringBuffer>);

**Comment by jakesmith** [comment]
@richardkchapman - a couple of comments.

File: roxie/ccd/ccdfile.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jakesmith) [reply roxie/ccd/ccdfile.cpp]: right yes, misread.

**Comment by jakesmith** [comment]
@richardkchapman - looks good afaics.

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

**Pull request merged**